src/main/kotlin/com/hibob/kotlinBasics/lambdaWithReceiver.kt

```diff
@@ -0,0 +1,111 @@
+package com.hibob.kotlinBasics
+import java.time.LocalDate
+
+
+/*
+1. Make the Runner.init function more readable using Lambda with Receiver function usage
+2. Initiate the variable movie in line 18 with the function createGoodMovie()
+3. Implement pretty print using buildString function
+4. Make SpidermanMovieProduceActions nullable (if not so yet) and make the relevant adjustments
+*/
+
+fun main() {
+    val movie = createGoodMovie()
+    val runner = Runner(movie)
+    val success = runner.init()
+
+    printSuccessMessage(success)
+    println("Pretty print: ${movie.prettyPrint()}")
+//    println("Json: ${movie.toJson()}")
+}
+
+fun printSuccessMessage(success: Boolean) {
+    if (success) {
+        println("The movie was successful")
+    } else {
+        println("The movie failed")
+    }
+}
+fun createGoodMovie():SpidermanNoWayHome{
+    return SpidermanNoWayHome()
+
+}
+interface SpidermanMovieProduceActions {
+    fun signTobeyMaguire()
+    fun signAndrew()
+    fun signTom()
+    fun getVillains()
+    fun isThereLockdown(): Boolean
+    fun publish():Boolean
+
+    val title: String
+    val airDate: LocalDate
+    val imdbRank: Double
+}
+
+class SpidermanNoWayHome() : SpidermanMovieProduceActions {
+
+    override val title: String = "Spiderman - No Way Home"
+    override val airDate: LocalDate = LocalDate.of(2021,12,16)
+    override val imdbRank: Double = 9.6
+
+    fun prettyPrint(): String = buildString{
+        append("Title: $title Air Date: $airDate IMDB rank: $imdbRank")
```

Comment:
Here you can do it more readable
there is an option to append each line separately